### PR TITLE
Add Brand faker

### PR DESCRIPTION
**Pull request by HattoriHenzo**
Add sport, car and watch brand with Brand faker. Generate random sport wearing brand, car brand or watch brand. Only generate brand by types of products.

### CODE REVIEW

File: src/main/java/net/datafaker/providers/base/BaseProviders.java [hunk 1]
```diff
@@ -65,6 +65,10 @@ default Business business() {
         return getProvider(Business.class, Business::new);
     }
 
+    default Brand brand() {
```
Comment (snuyanzin) [inline src/main/java/net/datafaker/providers/base/BaseProviders.java:68]: We prefer abc order in this interface

File: src/test/java/net/datafaker/providers/base/BrandTest.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+import org.junit.jupiter.api.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+class BrandTest extends BaseFakerTest<BaseFaker> {
+
+    @Test
+    void testSport() {
+        assertThat(faker.brand().sport()).isNotEmpty();
```
Comment (snuyanzin) [inline src/test/java/net/datafaker/providers/base/BrandTest.java:11]: It's better to follow the new way for tests like e.g. done within this method https://github.com/datafaker-net/datafaker/blob/34c43b81889077d679589be8adef8102d78d9acb/src/test/java/net/datafaker/providers/base/AviationTest.java#L32-L37

File: src/main/java/net/datafaker/providers/sport/SportProviders.java [hunk 3]
```diff
@@ -30,5 +30,4 @@ default Formula1 formula1() {
     default Volleyball volleyball() {
         return getProvider(Volleyball.class, Volleyball::new);
     }
-
```
[the remark this answers is not in the document]
Comment (kingthorin) [reply src/main/java/net/datafaker/providers/sport/SportProviders.java:33]: Still :wink:

File: src/main/resources/en/brand.yml [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+      watch:
+        - Armani Exchange
+        - Calvin Klein
+        - Casio
+        - Diesel
+        - Guess
+        - Mathey-Tissot
+        - Omega
+        - Rolex
+        - Swatch
```
[the remark this answers is not in the document]
Comment (kingthorin) [reply src/main/resources/en/brand.yml:35]: This should probably be alpha sorted as well. So Timex last.

File: src/test/java/net/datafaker/providers/base/BrandTest.java [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+
+    @Test
+    void testCar() {
+        assertThat(faker.brand().car()).isNotEmpty();
+    }
+
+    @Test
+    void testWatch() {
+        assertThat(faker.brand().watch()).isNotEmpty();
+    }
```
Comment (kingthorin) [inline src/test/java/net/datafaker/providers/base/BrandTest.java:25]: These can all be removed.

`providerListTest` takes care of it.

File: src/main/resources/en/brand.yml (hunk 4 above)
[the remark this answers is not in the document]
Comment (HattoriHenzo) [reply src/main/resources/en/brand.yml:35]: Thanks

File: src/main/java/net/datafaker/providers/sport/SportProviders.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (HattoriHenzo) [reply src/main/java/net/datafaker/providers/sport/SportProviders.java:33]: Want me to delete it?

[the remark this answers is not in the document]
Comment (kingthorin) [reply src/main/java/net/datafaker/providers/sport/SportProviders.java:33]: You should restore the blank line that was removed. No point committing a white space change 😉

[the remark this answers is not in the document]
Comment (HattoriHenzo) [reply src/main/java/net/datafaker/providers/sport/SportProviders.java]: Thanks

**Review by kingthorin — APPROVED** [review]

**Comment by bodiam** [comment]
Looks good to me now! Thanks for adding this and fixing the PR feedback!

**Pull request merged**